support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -137,7 +137,11 @@ public HttpResponse doSubmit(StaplerRequest request) throws IOException, Servlet
     @RequirePOST
     public HttpResponse doProceed(StaplerRequest request) throws IOException, ServletException, InterruptedException {
         preSubmissionCheck();
-
+        User user = User.current();
+        if (user!=null){
```

Comment:
If you are not adding an `ApproverAction` when the approver is anonymous (which I think is fine), then `ApproverAction` can declare `userId` as `@Nonnull` and skip the null checks, simplifying several things.